### PR TITLE
Add comments to the .c source file under rt-thread\components\libc #7426

### PR DESCRIPTION
**Pull request by shu-qian**
## 拉取/合并请求描述：(PR description)

[
The purpose of this PR is to submit the first compliant PR to complete the novice task and add comments in the source code to improve the readability and understandability of the code and contribute to the project.

I have added detailed comments to the functions, structures and variables in the .c source code under rt-thread\components\libc, explaining their functions, usage and precautions.

This PR is mainly for comments on the source code and does not involve specific functional changes, so there is no specific testing environment.
]

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: components/libc/compilers/armlibc/syscalls.c [hunk 1]
```diff
@@ -183,19 +156,10 @@ int _sys_read(FILEHANDLE fh, unsigned char *buf, unsigned len, int mode)
...
 
-/*
- * Write to a file. Returns 0 on success, negative on error, and
- * the number of characters _not_ written on partial success.
- * `mode' exists for historical reasons and must be ignored.
- * The return value is either:
- * A positive number representing the number of characters not written
- * (so any nonzero return value denotes a failure of some sort).
- * A negative number indicating an error.
- */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:198]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 2]
```diff
@@ -244,13 +208,6 @@ int _sys_write(FILEHANDLE fh, const unsigned char *buf, unsigned len, int mode)
     }
 }
 
-/*
- * Flush any OS buffers associated with fh, ensuring that the file
- * is up to date on disk. Result is >=0 if OK, negative for an
- * error.
- * This function is deprecated. It is never called by any other library function,
- * and you are not required to re-implement it if you are retargeting standard I/O (stdio).
- */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:253]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 3]
```diff
@@ -261,10 +218,6 @@ int _sys_ensure(FILEHANDLE fh)
 #endif /* DFS_USING_POSIX */
 }
 
-/*
- * Move the file position to a given offset from the file start.
- * Returns >=0 on success, <0 on failure.
- */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:267]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 4]
```diff
@@ -279,9 +232,6 @@ int _sys_seek(FILEHANDLE fh, long pos)
 #endif /* DFS_USING_POSIX */
 }
 
-/**
- * used by tmpnam() or tmpfile()
- */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:284]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 5]
```diff
@@ -297,32 +247,25 @@ int _sys_tmpnam(char *name, int fileno, unsigned maxlength)
 
 char *_sys_command_string(char *cmd, int len)
 {
-    /* no support */
+    /* No support. */
     return RT_NULL;
 }
 
-/* This function writes a character to the console. */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:304]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 6]
```diff
@@ -297,32 +247,25 @@ int _sys_tmpnam(char *name, int fileno, unsigned maxlength)
...
 void _ttywrch(int ch)
 {
 #ifdef RT_USING_CONSOLE
     rt_kprintf("%c", (char)ch);
 #endif /* RT_USING_CONSOLE */
 }
 
-/* for exit() and abort() */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:312]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 7]
```diff
@@ -297,32 +247,25 @@ int _sys_tmpnam(char *name, int fileno, unsigned maxlength)
...
 rt_weak void _sys_exit(int return_code)
 {
     extern void __rt_libc_exit(int status);
     __rt_libc_exit(return_code);
-    while (1);
+    while (1)
+        ;
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:266]: don't need to change for this

File: components/libc/compilers/armlibc/syscalls.c [hunk 8]
```diff
@@ -297,32 +247,25 @@ int _sys_tmpnam(char *name, int fileno, unsigned maxlength)
...
 }
 
-/**
- * return current length of file.
- *
- * @param fh - file handle
- * @return file length, or -1 on failed
- */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:325]: Please don't delete it.

File: components/libc/compilers/armlibc/syscalls.c [hunk 9]
```diff
@@ -366,7 +309,7 @@ int fputc(int c, FILE *f)
     rt_kprintf("%c", (char)c);
     return 1;
 #else
-    return 0; /* error */
+    return 0;     /* error */
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:312]: don't need to change

File: components/libc/compilers/armlibc/syscalls.c [hunk 10]
```diff
@@ -387,5 +330,4 @@ int fgetc(FILE *f)
     LOG_W("%s: %s", __func__, _WARNING_WITHOUT_STDIO);
     return 0; /* error */
 }
-
```
Comment (mysterywolf) [inline components/libc/compilers/armlibc/syscalls.c:390]: need a empty line, don't need to change

File: components/libc/compilers/common/cstring.c [hunk 11]
```diff
@@ -111,24 +175,52 @@ void* memrchr(const void* ptr, int ch, size_t pos)
...
+ * @param    maxlen is the maximum number of characters to consider.
+ *
+ * @return   The length of the string, up to a maximum of maxlen.
+ */
 size_t strnlen(const char *s, size_t maxlen)
 {
     const char *sc;
-    for (sc = s; maxlen != 0 && *sc != '\0'; maxlen--, ++sc);
+    for (sc = s; maxlen != 0 && *sc != '\0'; maxlen--, ++sc)
+        ;
```
Comment (mysterywolf) [inline components/libc/compilers/common/cstring.c:190]: let the ';' stay with the same line with for loop

File: components/libc/compilers/common/cstring.c [hunk 12]
```diff
@@ -202,14 +329,16 @@ rt_weak char *strtok_r(char *str, const char *delim, char **saveptr)
         return NULL;
     }
 
-    for (;*pbegin && strchr(delim, *pbegin) != NULL; pbegin++);
+    for (; *pbegin && strchr(delim, *pbegin) != NULL; pbegin++)
+        ;
```
Comment (mysterywolf) [inline components/libc/compilers/common/cstring.c:333]: one line

File: components/libc/compilers/common/cstring.c [hunk 13]
```diff
@@ -202,14 +329,16 @@ rt_weak char *strtok_r(char *str, const char *delim, char **saveptr)
...
 
     if (!*pbegin)
     {
         return NULL;
     }
 
-    for (pend = pbegin + 1; *pend && strchr(delim, *pend) == NULL; pend++);
+    for (pend = pbegin + 1; *pend && strchr(delim, *pend) == NULL; pend++)
+        ;
```
Comment (mysterywolf) [inline components/libc/compilers/common/cstring.c:341]: one line

File: components/libc/compilers/common/ctime.c [hunk 14]
```diff
@@ -43,28 +43,28 @@
...
-    (31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30 + 31 + 30),
-    (31 + 28 + 31 + 30 + 31 + 30 + 31 + 31 + 30 + 31 + 30 + 31),
+    {
+        0,
+        (31),
+        (31 + 28),
+        (31 + 28 + 31),
+        (31 + 28 + 31 + 30),
+        (31 + 28 + 31 + 30 + 31),
+        (31 + 28 + 31 + 30 + 31 + 30),
```
Comment (mysterywolf) [inline components/libc/compilers/common/ctime.c:61]: this format is good, don't need to change it

File: components/libc/compilers/common/ctime.c [hunk 15]
```diff
@@ -191,7 +191,8 @@ struct tm *gmtime_r(const time_t *timep, struct tm *r)
         work -= 1;
     }
 
-    for (i = 11; i && (__spm[i] > work); --i);
+    for (i = 11; i && (__spm[i] > work); --i)
+        ;
```
Comment (mysterywolf) [inline components/libc/compilers/common/ctime.c:195]: one line

File: components/libc/compilers/common/cwchar.c [hunk 16]
```diff
@@ -47,47 +56,25 @@ static int bisearch(wchar_t ucs, const struct interval *table, int max)
...
-        { 0x07A6, 0x07B0 }, { 0x0901, 0x0902 }, { 0x093C, 0x093C },
-        { 0x0941, 0x0948 }, { 0x094D, 0x094D }, { 0x0951, 0x0954 },
-        { 0x0962, 0x0963 }, { 0x0981, 0x0981 }, { 0x09BC, 0x09BC },
-        { 0x09C1, 0x09C4 }, { 0x09CD, 0x09CD }, { 0x09E2, 0x09E3 },
-        { 0x0A02, 0x0A02 }, { 0x0A3C, 0x0A3C }, { 0x0A41, 0x0A42 },
-        { 0x0A47, 0x0A48 }, { 0x0A4B, 0x0A4D }, { 0x0A70, 0x0A71 },
-        { 0x0A81, 0x0A82 }, { 0x0ABC, 0x0ABC }, { 0x0AC1, 0x0AC5 },
-        { 0x0AC7, 0x0AC8 }, { 0x0ACD, 0x0ACD }, { 0x0B01, 0x0B01 },
-        { 0x0B3C, 0x0B3C }, { 0x0B3F, 0x0B3F }, { 0x0B41, 0x0B43 },
-        { 0x0B4D, 0x0B4D }, { 0x0B56, 0x0B56 }, { 0x0B82, 0x0B82 },
```
Comment (mysterywolf) [inline components/libc/compilers/common/cwchar.c:69]: this format is good , don't need to change it.

File: components/libc/compilers/common/cwchar.c [hunk 17]
```diff
@@ -97,35 +84,43 @@ int wcwidth(wchar_t ucs)
...
+             (ucs >= 0xf900 && ucs <= 0xfaff) || /* CJK Compatibility Ideographs */
+             (ucs >= 0xfe30 && ucs <= 0xfe6f) || /* CJK Compatibility Forms */
+             (ucs >= 0xff00 && ucs <= 0xff5f) || /* Fullwidth Forms */
+             (ucs >= 0xffe0 && ucs <= 0xffe6)    // ||
+                                                 // #ifndef _WIN32
+             //       (ucs >= 0x20000 and ucs <= 0x2ffff)
+             // #else
+             //     0
+             // #endif
+             ));
```
Comment (mysterywolf) [inline components/libc/compilers/common/cwchar.c:108]: please make sure the format.

File: components/libc/compilers/dlib/syscall_close.c [hunk 18]
```diff
@@ -5,30 +5,42 @@
...
+ * @return  0 on success, or a nonzero value on failure.
+ *
+ * @note    If the 'handle' corresponds to standard input (stdin), standard output (stdout), or standard error (stderr),
+ *          this function returns an error. Otherwise, it attempts to close the file handle using the 'close' function.
+ *          If RT-Thread's POSIX file system (DFS_USING_POSIX) is enabled, it delegates the close operation to the 'close' function,
+ *          otherwise, it logs a warning and returns an error.
  */
-
 #pragma module_name = "?__close"
-
```
Comment (mysterywolf) [inline components/libc/compilers/dlib/syscall_close.c:25]: please don't delte the space.

File: components/libc/compilers/dlib/syscall_lseek.c [hunk 19]
```diff
@@ -5,39 +5,44 @@
...
 
 #include <rtthread.h>
 #include <LowLevelIOInterface.h>
 #include <unistd.h>
 #include <compiler_private.h>
-#define DBG_TAG    "dlib.syscall.lseek"
-#define DBG_LVL    DBG_INFO
+
+#define DBG_TAG "dlib.syscall.lseek"
+#define DBG_LVL DBG_INFO
```
Comment (mysterywolf) [inline components/libc/compilers/dlib/syscall_lseek.c:17]: don't need to change

**Review by mysterywolf — CHANGES_REQUESTED** [review]
Hi, Thanks for submit the PR! there are too many code in this PR, so that it will be painful to review it. Could you split to several PRs? the first PR I think is  you can add some comments of functions for the sources files. Once you've already been familar with rt-thread's code format style, then you may submit a banch of code in one PR. I just review part of them.

**Comment by shu-qian** [comment]
ok, I will divide it into several PR submissions, thank you for your review. I am sorry that there were some unintentional inappropriate modifications due to too many files.

**Pull request closed without merging**